services/infra/InfraService.ts

```diff
@@ -11,41 +12,87 @@ interface InfraServiceProps {
   sitesService: SitesService
   reposService: ReposService
   deploymentsService: DeploymentsService
+  mailer: MailClient
 }
 
 export default class InfraService {
-  private readonly usersService: InfraServiceProps["usersService"]
+  private readonly usersService
 
-  private readonly sitesService: InfraServiceProps["sitesService"]
+  private readonly sitesService
 
-  private readonly reposService: InfraServiceProps["reposService"]
+  private readonly reposService
 
-  private readonly deploymentsService: InfraServiceProps["deploymentsService"]
+  private readonly deploymentsService
+
+  private readonly mailer
 
   constructor({
     usersService,
     sitesService,
     reposService,
     deploymentsService,
+    mailer,
   }: InfraServiceProps) {
     this.usersService = usersService
     this.sitesService = sitesService
     this.reposService = reposService
     this.deploymentsService = deploymentsService
+    this.mailer = mailer
+  }
+
+  sendCreateError = async (
+    email: string,
+    repoName: string | undefined,
+    submissionId: string | undefined,
+    error: string
```

Comment:
i'm not sure if there's a way around this but the intent of this `sendCreateError` seems to be to provide a diagnostic for end users. hence, i'm not sure that passing `undefined | string` would be a good idea here. i think a better API here would either be to force devs to specify `repoName/submissionId` or if unable, provide a default argument/alternative text in the function body so that our users still get a useful diagnostic

Reply:
This problem went away when I moved FormSG-related stuff out of InfraService and into formsgSiteCreation.